util/oidc/oidc.go

```diff
@@ -288,6 +288,7 @@ func (a *ClientApp) HandleLogin(w http.ResponseWriter, r *http.Request) {
 	returnURL := r.FormValue("return_url")
 	// Check if return_url is valid, otherwise abort processing (see https://github.com/argoproj/argo-cd/pull/4780)
 	if !isValidRedirectURL(returnURL, []string{a.settings.URL}) {
+		log.Errorf("Invalid redirect URL: %s is not a valid URL or does not match the URL in settings.", returnURL)
```

Comment:
Does this happen before or after auth? If before, I think we should truncate the returnURL to some reasonable length and remove any newlines to prevent dumping a bunch of bad stuff in logs.

Reply:
It's before auth.  Called `text.Trunc` and `strings.Replace` on `returnURL`.

Just as an aside, there's a `truncateString` function in `cmd/argocd/commands/app.go` and a `Trunc` function in `util/text/text.go`.  Maybe the former is redundant?